src/utils/polygon.cpp

```diff
@@ -360,6 +369,49 @@ Polygons Polygons::offset(int distance, ClipperLib::JoinType join_type, double m
     return ret;
 }
 
+Polygons Polygons::offset(const std::vector<int>& offset_dists) const
+{
+    // we need as many offset-dists as points
+    assert(this->pointCount() == offset_dists.size());
+
+    Polygons ret;
+    int i = 0;
+    for (auto& poly_line : this->paths)
```

Comment:
```suggestion
    for (auto& poly_line : this->paths | ranges::views::filter([](const auto& path){ return ! path.empty(); }))
```

If there is an empty `path` in this collection of `paths` it would crash at line 384. Don't forget to add `#include <range/v3/views/filter.hpp>` at the top